tests/memoryview/memoryview_annotation_typing.py

```diff
@@ -37,6 +42,30 @@ def two_dim(a: cython.double[:,:]):
     a[0,0] *= 3
     return a[0,0], a[0,1], a.ndim
 
+def slice_none(m: cython.double[:]):
+    """
+    >>> try:
+    ...     a = slice_none(None)
+    ... except TypeError as exc:
+    ...     assert COMPILED
+    ...     if "Argument 'm' must not be None" not in str(exc): raise
+    ... else:
+    ...     assert a == 1
+    ...     assert not COMPILED
+    """
+    return 1 if m is None else 2
+
+
+def slice_optional(m: typing.Optional[cython.double[:]]):
+    """
+    >>> slice_optional(None)
+    1
+    >>> a = numpy.ones((10,), numpy.double)
+    >>> slice_optional(a)
+    2
+    """
```

Comment:
```suggestion

    # Make sure that we actually evaluate the type and don't just accept everything.
    >>> try: slice_optional(123)
    ... except ValueError:
    ...     assert COMPILED
    ... else:
    ...     assert not COMPILED
    """
```

Reply:
> Let me try that again. What about this test?

`ValueError` is not raised but `TypeError`. I have pushed fixed and passing version. Have a look. I hope now it is fine.